tree/30_generic_methods/directory_absent.cf

```diff
@@ -0,0 +1,79 @@
+#####################################################################################
+# Copyright 2017 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Directory absent
+# @description Ensure a directory's absence
+# @documentation If `recursive` is false, only an empty directory can be deleted.
+#
+# @parameter target     Directory to remove
+# @parameter recursive  Should deletion be recursive, "true" or "false" (defaults to "false")
+# @parameter_constraint recursive "allow_empty_string" : true
+#
+# @class_prefix directory_absent
+# @class_parameter target
+
+bundle agent directory_absent(target, recursive)
+{
+  vars:
+      "class_prefix"            string => canonify("directory_absent_${target}");
+
+  defaults:
+      "recursive"               string => "false", if_match_regex => "";
+
+  classes:
+      "is_recursive"        expression => strcmp("true", "${recursive}");
+
+    pass2::
+      "target_exists_after" expression => fileexists("${target}");
```

Comment:
shouldn't you use isdir to avoid detecting file that would have the same name as folder?
https://docs.cfengine.com/docs/3.7/reference-functions-isdir.html